lib/item_repository.rb

```diff
@@ -0,0 +1,39 @@
+require 'item'
+
+class ItemRepository
+  
+  def all
+    arr_items = []
+    sql = 'SELECT id, name, unit_price, item_quantity FROM items;'
+    result_set = DatabaseConnection.exec_params(sql, arr_items)
+
+    result_set.each do |record|
+      item = Item.new
+
+      item.id = record['id'].to_i
+      item.name = record['name']
+      item.unit_price = record['unit_price'].to_i
+      item.item_quantity = record['item_quantity'].to_i
+  
+      arr_items << item
+    end
+  return arr_items
```

Comment:
'all' method is efficient and utilises only the necessary code to return all items